tests/datasets.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+from sknnr.datasets import load_moscow_stjoes
+
+
+@dataclass
+class TestDataset:
+    __test__ = False
+
+    X_train: np.ndarray
+    X_test: np.ndarray
+    y_train: np.ndarray
+    y_test: np.ndarray
+    ref_distances: np.ndarray
+    ref_neighbors: np.ndarray
+    trg_distances: np.ndarray
+    trg_neighbors: np.ndarray
+    trg_predicted_weighted: np.ndarray
+    trg_predicted_unweighted: np.ndarray
+
+
+def load_moscow_stjoes_results(method: str, k: int = 5) -> TestDataset:
+    """Load the Moscow Mountain / St. Joe's dataset results for port testing.
+
+    The dataset will always be in dataframe format, with a fixed 80% train/test split
+    applied. New attributes are added with results from yaImpute using the given method
+    and split.
+    """
+    X, y = load_moscow_stjoes(return_X_y=True, as_frame=True)
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, train_size=0.8, shuffle=False
+    )
+
+    ref_distances = pd.read_csv(f"./tests/data/{method}_moscow_ref_distances_k{k}.csv")
+    ref_neighbors = pd.read_csv(f"./tests/data/{method}_moscow_ref_neighbors_k{k}.csv")
+    trg_distances = pd.read_csv(f"./tests/data/{method}_moscow_trg_distances_k{k}.csv")
+    trg_neighbors = pd.read_csv(f"./tests/data/{method}_moscow_trg_neighbors_k{k}.csv")
+    trg_predicted_weighted = pd.read_csv(
+        f"./tests/data/{method}_moscow_trg_predicted_weighted_k{k}.csv"
+    )
+    trg_predicted_unweighted = pd.read_csv(
+        f"./tests/data/{method}_moscow_trg_predicted_unweighted_k{k}.csv"
+    )
```

Comment:
I do like being able to use an absolute module path, but I would lean towards sticking with file paths just so that we can keep the convenience of `pandas`. Otherwise, we would need to add a utility function to load the CSVs as dataframes with columns (similar to `sknnr.datasets._base._load_csv_data`, but with a few modifications). Let me know if you disagree!

Reply:
It's fine by me to keep it as is.  But I'm confused why this approach wouldn't work?
```
import pandas as pd
from sknnr.datasets._base import _open_text
DATA_MODULE = "sknnr.datasets.data"
with _open_text(DATA_MODULE, "moscow_env.csv") as fh:
    df = pd.read_csv(fh)
```
This is obviously an example using `moscow_env.csv` because our test data is outside the `sknnr` package.  In order for this to work with the test data, we'd have to put it in the package which may be distasteful.  Sorry for the naive question!

Reply:
You're right, I didn't realize that `pd.read_csv` could open a buffer! This would totally work.

> In order for this to work with the test data, we'd have to put it in the package which may be distasteful.

It looks like this works if we use `DATA_MODULE = "tests.data"`. I think we'll just need to add an `__init__.py` to that directory for compatibility with 3.8 (like I had to do in 36f37b1b47738777566736ffc9fee621fc4d3445).

I'll make the change!

Reply:
Nice!  I didn't look at `importlib.resources` carefully enough to know that it could pull from anywhere in the package.